addons/rearm/functions/fnc_rearm.sqf

```diff
@@ -27,11 +27,16 @@ if (isNil "_magazineClass") exitWith {ERROR_1("magazineClass nil",_attachedDummy
 ([_magazineClass] call FUNC(getCaliber)) params ["_cal", "_idx"];
 
 // Get magazines that can be rearmed
-private _needRearmMags = [_target, _magazineClass] call FUNC(getNeedRearmMagazines);
-_needRearmMags params ["_needRearm", "_turretPath", "_cnt", "_pylon"];
+private _needRearmMags = [_target] call FUNC(getNeedRearmMagazines);
+private _needRearmMagsOfClass = _needRearmMags select {(_x select 0) isEqualTo _magazineClass};
 
 // Exit if no magazines need rearming
-if (!_needRearm) exitWith {ERROR_2("Could not find turret for %1 in %2", _magazineClass, typeOf _target);};
+if ((count _needRearmMagsOfClass) == 0) exitWith {ERROR_2("Could not find turret for %1 in %2", _magazineClass, typeOf _target);};
+
+private _currentRearmableMag = _needRearmMagsOfClass select 0;
+private _turretPath = _currentRearmableMag select 1;
+private _pylon = _currentRearmableMag select 3;
+private _magazineCount = _currentRearmableMag select 5;
```

Comment:
It's not big I assume, since it doesn't do anything heavy. I'd say readability is better with `params` myself, but up to you how you want to keep it.